### PR TITLE
Infrastructure and 1 E2E desktop test with Selenium

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: tests/Microsoft.Identity.Test.Integration/Infrastructure/SeleniumExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium.Support.UI;
+using System.Diagnostics;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.Identity.Test.Core.UIAutomation;
+
+namespace Microsoft.Identity.Test.Integration.Infrastructure
+{
+    public static class SeleniumExtensions
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Integration/Infrastructure/SeleniumExtensions.cs]: Most of this was moved as this code existed in a different assembly.

File: tests/Microsoft.Identity.Test.Integration/Infrastructure/SeleniumWebUI.cs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+using OpenQA.Selenium;
+using System;
+using System.Diagnostics;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Test.Integration.Infrastructure
+{
+    internal class SeleniumWebUI : IWebUI
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Integration/Infrastructure/SeleniumWebUI.cs]: This is the "orchestator" - it kicks off the selenium automation, it kicks off listening to TCP requests and it parses the URI for a result.

File: tests/Microsoft.Identity.Test.Core.UIAutomation/UserInformationFieldIds.cs [hunk 3]
```diff
@@ -26,52 +26,83 @@
...
 
 namespace Microsoft.Identity.Test.Core.UIAutomation
 {
     public class UserInformationFieldIds
     {
-        public string PasswordInputId { get; set; }
-        public string SignInButtonId { get; set; }
+        private readonly LabUser _user;
+        private string _passwordInputId;
+        private string _singInButtonId;
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Core.UIAutomation/UserInformationFieldIds.cs]: >ng [](start = 26, length = 2)

sp. _signInButtonId

File: tests/Microsoft.Identity.Test.Integration/Infrastructure/SeleniumWebUI.cs [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+                        Error = "system_browser_cancel_or_timeout_exception"
+                    };
+
+                    return await Task.FromResult(result).ConfigureAwait(false);
+                }
+                catch (SocketException ex)
+                {
+                    var result = new AuthorizationResult(AuthorizationStatus.UnknownError)
+                    {
+                        ErrorDescription = ex.Message + " soket error code: " + ex.SocketErrorCode,
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Integration/Infrastructure/SeleniumWebUI.cs]: >ok [](start = 59, length = 2)

sp...missing "c" :)

File: tests/Microsoft.Identity.Test.Integration/Infrastructure/SeleniumWebUI.cs [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+                        Error = "system_browser_soket_exception"
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Integration/Infrastructure/SeleniumWebUI.cs]: >k [](start = 50, length = 1)

sp.

**Comment by jennyf19** [comment]
> using OpenQA.Selenium;

do the files you added need the license header info?

---
Refers to: tests/Microsoft.Identity.Test.Integration/Infrastructure/SeleniumExtensions.cs:1 in 7f2d667. [](commit_id = 7f2d66767742230cac88fec3ebae5f4dff4472c3, deletion_comment = False)

File: tests/Microsoft.Identity.Test.Integration/Infrastructure/SingleMessageTcpListener.cs [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+            TcpClient tcpClient,
+            Func<Uri, string> responseProducer,
+            CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            string httpRequest = await GetTcpResponseAsync(tcpClient, cancellationToken).ConfigureAwait(false);
+            Uri uri = ExtractUriFromHttpRequest(httpRequest);
+
+            // AuthorizationResult authenticationResult = new AuthorizationResult(AuthorizationStatus.Success, uri);
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Integration/Infrastructure/SingleMessageTcpListener.cs]: is this supposed to be commented?

File: tests/Microsoft.Identity.Test.Integration/Infrastructure/SingleMessageTcpListener.cs [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+
+            return stringBuilder.ToString();
+        }
+
+        private async Task WriteResponseAsync(
+            string message,
+            NetworkStream stream,
+            CancellationToken cancellationToken)
+        {
+            //string message = null;
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Integration/Infrastructure/SingleMessageTcpListener.cs]: is this needed?

File: tests/Microsoft.Identity.Test.Integration/SeleniumTests/InteractiveFlowTests.cs [hunk 8]
```diff
@@ -0,0 +1,82 @@
+﻿using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Core;
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Integration/SeleniumTests/InteractiveFlowTests.cs]: license info?

**Comment by henrik-me** [comment]
> <wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">

I believe Mark was using these... though I'm fine these are not here.

---
Refers to: LibsAndSamples.sln.DotSettings:1 in 7f2d667. [](commit_id = 7f2d66767742230cac88fec3ebae5f4dff4472c3, deletion_comment = True)

File: src/Microsoft.Identity.Client/WsTrust/UsernamePasswordInput.cs [hunk 9]
```diff
@@ -1,4 +1,4 @@
-//------------------------------------------------------------------------------
+    //------------------------------------------------------------------------------
```
Comment (henrik-me) [inline src/Microsoft.Identity.Client/WsTrust/UsernamePasswordInput.cs]: >    [](start = 1, length = 3)

revert?

**Pull request merged**